protected/humhub/modules/space/widgets/CustomSidebar.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+
+namespace humhub\modules\space\widgets;
+
+class CustomSidebar extends Sidebar
```

Comment:
If I understand it correctly, this is the sidebar of the About page? Can we call it AboutPageSidebar then? Or am I missing something here?

Reply:
Here it's similar to the menu, the modules themselves have to be so smart and only add themselves to the sidebar when it makes sense.

I think an additional AboutPageSidebar class would be best. Here we can display the two widgets (MyMembership / Members). MyMembership of course only for members. And if a module (e.g. statistics) wants to add a widget here, it can do so. 

What do you think?

Reply:
Currently, we use the `Sidebar` widget like this : 
````php
$this->beginBlock('sidebar'); ?>
<?= Sidebar::widget(['space' => $space, 'widgets' => [
    [MyMembership::class, ['space' => $space], ['sortOrder' => 10]],
    [Members::class, ['space' => $space], ['sortOrder' => 20]]
]]);
?>
<?php $this->endBlock(); ?>
````
As you can see, it's ok to choose which widgets be in it.
The problem is about pre-added widgets like Events, that I think we can handle it by a new attribute in the `Sidebar` widget. We can simply add the below codes in the `init()` function of the `Sidebar` widget: 
````php
if (!$this->preAddedWidgets)
    $this->widgets = [];
````

and use it like this: 
````php
$this->beginBlock('sidebar'); ?>
<?= Sidebar::widget(['space' => $space, 'preAddedWidgets' => false, 'widgets' => [
    [MyMembership::class, ['space' => $space], ['sortOrder' => 10]],
    [Members::class, ['space' => $space], ['sortOrder' => 20]]
]]);
?>
<?php $this->endBlock(); ?>
````

Reply:
I named it `CustomSidebar` instead of `AboutPageSidebar` because I think maybe in the future we want to use it again on other pages.
As its usage is like below, so we can choose which widgets be in it.
````php
<?php
$this->beginBlock('sidebar'); ?>
<?= CustomSidebar::widget(['space' => $space, 'baseWidgets' => false, 'customWidgets' => [
    [MyMembership::class, ['space' => $space], ['sortOrder' => 10]],
    [Members::class, ['space' => $space], ['sortOrder' => 20]]
]]);
?>
<?php $this->endBlock(); ?>
````

Reply:
Okay, got it.

Since we work a lot with events and especially with widgets it is probably easier if you (a module developer) can listen to class level events like an AboutPageSidebar::EVENT_INIT -  so I would prefer a own widget class for the abotu page. 

You could also use the init method to check or the user is a member and only in this case add the MyMembership widget. 

What do you think about it?